server/src/main/java/org/elasticsearch/search/aggregations/AggregationExecutionContext.java

```diff
@@ -53,6 +53,12 @@ public long getTimestamp() {
     }
 
     public int getTsidOrd() {
+        if (tsidOrdProvider == null) {
+            throw new AggregationExecutionException(
```

Comment:
I think this will return a 500 status code. Maybe throw an illegal argument exception? this should return a 400 status code.